Realm/RLMRealm.mm

```diff
@@ -453,7 +475,12 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration
     }
 
     // First check if we already have a cached Realm for this thread/config
-    if (auto realm = getCachedRealm(configuration, cacheKey)) {
+    RLMRealm *cachedRealm = getCachedRealm(configuration, cacheKey);
+    // Check if this is the first time opening the realm or not.
+    bool isFirstOpen = (cachedRealm == nil) ? true : false;
+
+    if (auto realm = cachedRealm) {
+        [realm subscribeToInitialSubscriptionsWithConfiguration:configuration isFirstOpen:isFirstOpen];
         return RLMAutorelease(realm);
     }
```

Comment:
```suggestion
    if (auto realm = getCachedRealm(configuration, cacheKey)) {
        [realm subscribeToInitialSubscriptionsWithConfiguration:configuration isFirstOpen:NO];
        return RLMAutorelease(realm);
    }
```

Because the `getCachedRealm` will return an instance of a cached Realm here to don't need the extra checks